lib/ehmdmc_client.rb

```diff
@@ -2,30 +2,123 @@
 
 module EHMDMCClient
 
-  def self.validate?(hmdmc)
-    r = connection.get('?hmdmc=' + sanitise(hmdmc))
-    unless r.status==200
-      # Something went wrong
-      Rails.logger.error "Attempting to validate HMDMC #{hmdmc} returned status code #{r.status}"
-      return false
+  class Validation
+    # This class will store the message for the error produced by the HMDMC service provider, 
+    # depending on the type of error it will generate a different content. Its purpose is to solve
+    # the following problems:
+    # - Publish the error message to the Logging facility to be able to track the problem
+    # - Provide to the user a meaningful text indicating the problem but without giving too much
+    # details about the internal process
+    # - Depending on the type of error, be able to know if the validation already happened (is_validated)
+    # or if we would need to retry in future, because it has not been validated
+    # - Indicate, when the validation has been performed, if the hmdmc is considered valid or not
+    # - Generate a valid json message for the client to display the relevant information for the user
+
+    attr_reader :error_message, :hmdmc, :text
+
+    def initialize(hmdmc)
+      @hmdmc = hmdmc
+      reset
+    end
+
+    def reset
+      @validated = true
+      @valid = true      
+    end
+
+    def is_validated?
+      @validated
+    end
+
+    def valid?
+      @validated && @valid
     end
+
+    def log
+      Rails.logger.send(@facility, @text) unless @text.nil?
+    end
+
+    def load_message(type_of_message, facility, text, validated)
+      not_validated_msg = "The HMDMC #{hmdmc} is considered as not valid by the HMDMC service"
```

Comment:
Very minor: `hmdmc` shows with the underscore instead of slash when shown in the error message